src/bag.test.js

```diff
@@ -176,6 +176,29 @@ describe("rosbag - high-level api", () => {
     );
   });
 
+  it("reads messages from a shuffled bag", async () => {
+    const topics = [
+      "/cloud_nodelet/parameter_descriptions",
+      "/cloud_nodelet/parameter_updates",
+      "/diagnostics",
+      "/gps/fix",
+      "/gps/rtkfix",
+      "/gps/time",
+      "/obs1/gps/rtkfix",
+      "/obs1/gps/time",
+      "/radar/range",
+      "/radar/tracks",
+      "/rosout",
+      "/tf",
+    ];
+    const messages = await fullyReadBag("demo-shuffled", {
+      topics,
+      noParse: true,
+      endTime: { sec: 1490148912, nsec: 600000000 },
+    });
+    expect(messages).toHaveLength(9);
```

Comment:
maybe worth asserting on message order, so that we're sure that one message goes backwards in time from the previous one?